frontend/views/containers/payments/PaymentsMixin.js

```diff
@@ -12,14 +12,75 @@ const PaymentsMixin: Object = {
     ...mapState(['currentGroupId']),
     ...mapGetters([
       'currentGroupState',
+      'dueDateForPeriod',
+      'groupPeriodPayments',
       'ourUsername',
       'ourPayments',
-      'groupPeriodPayments',
-      'paymentHashesForPeriod',
-      'dueDateForPeriod'
+      'periodAfterPeriod',
+      'periodBeforePeriod',
+      'periodStampGivenDate',
+      'paymentHashesForPeriod'
     ])
   },
   methods: {
+    async getPeriodStampGivenPayment (payment) {
+      return await this.getPeriodStampGivenDate(payment.date)
+    },
+
+    // Oldest key first.
+    async getSortedPeriodKeys () {
+      const key = `paymentsByPeriod/${this.ourUsername}/${this.currentGroupId}`
+      const archivedPaymentsByPeriod = await sbp('gi.db/archive/load', key) || {}
+      return [
+        ...Object.keys(archivedPaymentsByPeriod).sort(),
+        ...Object.keys(this.groupPeriodPayments).sort()
+      ]
+    },
+
+    // Note: 'recentDate' is a confusing name, as it can be in the future, or far in the past.
+    async getPeriodStampGivenDate (givenDate: string | Date) {
```

Comment:
Rename to `historicalPeriodStampGivenDate`

Reply:
It can also return non-historical results, so I think it would be a bit confusing

Reply:
True, but the point is that it unlike `periodStampGivenDate` it *can* give historical payment periods, so that's why it needs to be named that way. See also the [mental mode](https://github.com/okTurtles/group-income/pull/1691#discussion_r1327635564) comment I just posted.

Reply:
I'm also not fan of dropping the `get-` prefix here, as all other methods in this file are `get-` prefixed. It likely helps to distinguish them from computed properties. But I also think `getHistoricalPeriodStampGivenDate` would be a tad long...

Reply:
Should all `PaymentsMixin.js` methods that can return both in-memory and historical results be named that way (some currently don't)?

Reply:
Yeah feel free to make it consistent and clear 👍 
(I haven't checked your latest changes as of this comment but will have a look now)